### PR TITLE
Add compatibility for changes in repeater widget

### PR DESCRIPTION
**Pull request by SebastiaanKloos**
PR for https://github.com/octobercms/october/issues/3595

Trigger `change.oc.formwidget` when adding of removing repeater item

### CODE REVIEW

**Comment by SebastiaanKloos** [comment]
Tests so far on my end don't cause any trouble. I think it is ready to test!

File: modules/backend/widgets/form/assets/js/october.form.js [hunk 1]
```diff
@@ -117,13 +122,12 @@
...
         })
-
+        
         /*
          * When a master is updated, refresh its slaves
          */
         $.each(fieldMap, function(fieldName, toRefresh){
-            fieldElements.filter('[data-field-name="'+fieldName+'"]')
-                .on('change.oc.formwidget', $.proxy(self.onRefreshDependants, self, fieldName, toRefresh))
+            $(document).on('change.oc.formwidget', '[data-field-name="' + fieldName + '"]', $.proxy(self.onRefreshDependants, self, fieldName, toRefresh));
```
Comment (LukeTowers) [inline modules/backend/widgets/form/assets/js/october.form.js]: Can you explain this change more please?

File: modules/backend/widgets/form/assets/js/october.form.js [hunk 2]
```diff
@@ -137,6 +141,10 @@
             formEl = this.$form,
             fieldElements = this.getFieldElements()
 
+        if (!form) {
+            return;
+        }
```
Comment (LukeTowers) [inline modules/backend/widgets/form/assets/js/october.form.js]: Can you explain under what conditions this triggers?

File: modules/backend/widgets/form/assets/js/october.form.js [hunk 3]
```diff
@@ -83,8 +83,13 @@
             return this.fieldElementCache
         }
 
-        var form = this.$el,
-            nestedFields = form.find('[data-control="formwidget"] [data-field-name]')
+        var form = this.$el
+
+        if (!form) {
+            return;
+        }
```
Comment (LukeTowers) [inline modules/backend/widgets/form/assets/js/october.form.js]: Can you explain under what conditions this triggers?

File: modules/backend/widgets/form/assets/js/october.form.js (hunk 1 above)
Comment (SebastiaanKloos) [reply modules/backend/widgets/form/assets/js/october.form.js]: The repeater items can be added after the dom loaded. To add event listeners to items created after dom loaded you have to listen to document changes and filter it by field

File: modules/backend/widgets/form/assets/js/october.form.js (hunk 2 above)
Comment (SebastiaanKloos) [reply modules/backend/widgets/form/assets/js/october.form.js]: When adding item, then removing one, then adding one again. The form element cannot be found. Don't know why yet

File: modules/backend/widgets/form/assets/js/october.form.js (hunk 3 above)
Comment (SebastiaanKloos) [reply modules/backend/widgets/form/assets/js/october.form.js]: When adding item, then removing one, then adding one again. The form element cannot be found. Don't know why yet

Comment (SebastiaanKloos) [reply modules/backend/widgets/form/assets/js/october.form.js]: It fixed this
CleanShot 2019-11-05 at 22 32 33

File: modules/backend/widgets/form/assets/js/october.form.js (hunk 2 above)
Comment (SebastiaanKloos) [reply modules/backend/widgets/form/assets/js/october.form.js]: It fixed this
CleanShot 2019-11-05 at 22 32 33

**Comment by SebastiaanKloos** [comment]
Ping

**Comment by LukeTowers** [comment]
@SebastiaanKloos we're finalizing 460 right now, this will be reviewed for inclusion in 461

**Comment by SebastiaanKloos** [comment]
@LukeTowers, The ping was meant for the issue to be unarchived :-)

**Comment by LukeTowers** [comment]
@SebastiaanKloos gotcha.

**Comment by SebastiaanKloos** [comment]
This pull request needs to be reopened

File: modules/backend/widgets/form/assets/js/october.form.js (hunk 3 above)
Comment (LukeTowers) [reply modules/backend/widgets/form/assets/js/october.form.js]: I found the problem, you were removing the element that the event was triggered on before the change event could be fired on the containing form widget.

**Comment by LukeTowers** [comment]
Fixed in https://github.com/octobercms/october/commit/52cdfda93564723de684a0951c190d532c8f2dc7

**Pull request closed without merging**